src/SonarScanner.MSBuild.Common/EnvironmentBasedPlatformHelper.cs

```diff
@@ -25,7 +25,9 @@ namespace SonarScanner.MSBuild.Common;
 
 public class EnvironmentBasedPlatformHelper : IPlatformHelper
 {
+#if !NETSTANDARD1_1_OR_GREATER && !NETCOREAPP1_0_OR_GREATER
     private bool? isMacOs;
```

Comment:
Make `static`

Reply:
Turned it into a `Lazy` variable. The class is used as a singleton, so I didn't use a static variable there.